validator/client/validator.go

```diff
@@ -38,13 +40,42 @@ func (v *validator) Done() {
 }
 
 // WaitForActivation checks whether the validator pubkey is in the active
-// validator set. If not, this operation will block until an activation is
+// validator set. If not, this operation will block until an activation message is
 // received.
 //
 // WIP - not done.
 func (v *validator) WaitForActivation(ctx context.Context) {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "validator.WaitForActivation")
 	defer span.Finish()
+	// First, check if the beacon chain has started.
```

Comment:
Can all of the code for waiting for chain start go into the initialization method? That was the intention

Reply:
Also this makes the method do two things instead of one. Wait for activation should only call the method to wait for activation